### DISCUSSION TITLE
feat: Publish arm64 compatible docker images

### DISCUSSION THREAD

**Discussion by code-haven** [langfuse/langfuse#1066] [Ideas]
### Describe the feature you'd like to request

Would love to see arm64 compatible docker images that we can deploy to AWS graviton instances!

### Describe the solution you'd like to see

Building multi-arch docker images is quite easy with buildx: https://docs.docker.com/engine/reference/commandline/buildx/

### Contribute

- [x] Yes, I can implement this and raise a PR

**Comment by maxdeichmann** [comment]
@code-haven thanks. for raising the issue. Do you want to create a PR for this? Would you create multiple images for the different architectures or have one image which supports all of them?

**Comment by code-haven** [comment]
Hey, I'm happy to take this up. With multi-arch images, users can continue to use `ghcr.io/langfuse/langfuse:latest`. The docker client is able to pull the layers that are compatible with the host architecture.

**Comment by maxdeichmann** [comment]
> Hey, I'm happy to take this up. With multi-arch images, users can continue to use `ghcr.io/langfuse/langfuse:latest`. The docker client is able to pull the layers that are compatible with the host architecture.

Ok, awesome. Thanks already for the contribution.

**Comment by code-haven** [comment]
Ah, looks like buildx slows down the build from 3mins to  18mins due to the overhead from emulation. Not sure how I can speed this up :( 
https://github.com/code-haven/langfuse/actions/runs/6827802599/job/18570718484

**Comment by maxdeichmann** [comment]
> Ah, looks like buildx slows down the build from 3mins to 18mins due to the overhead from emulation. Not sure how I can speed this up :( https://github.com/code-haven/langfuse/actions/runs/6827802599/job/18570718484

Oh wow, this looks quite long. Do you know someone who is an expert in this and could help out?

**Comment by code-haven** [comment]
Nope. Reading up on this, it looks like we might need native arm64 github runners to avoid the emulation overhead

**Comment by marcklingen** [comment]
Is there a huge upside that justifies the effort? The Langfuse API is quite low-profile and should scale well on x86.

**Comment by code-haven** [comment]
nope, not at all. This is definitely a good to have at this point.

**Comment by verdverm** [comment]
My understanding going back and forth with the maintainer of the action (can't find it right now) was that there is an FS thing that is causing the majority of the pain. If I wanted to solve it best, I would adopt the arm runners or find a 3rd party provider for mac

With my own project, I resolved to
1. build arm/mac on important branches, skip on most
2. deal with the long runtimes

I'm seeing a warning message on Mac silicon, but have not noticed a perf issue that hinders development: https://github.com/orgs/langfuse/discussions/1460#discussioncomment-8862851

**Comment by marcklingen** [comment]
First version to include arm image was just released: https://github.com/langfuse/langfuse/releases/tag/v2.58.0
Thanks again @rennokki for the contribution
